### DISCUSSION TITLE
Migrating from Cocos2d-x 4.0 with SDKBox to adxe to axys

### DISCUSSION THREAD

**Discussion by qoozta** [axmolengine/axmol#833] [Q&A] [answered]
I've tested the latest dev commit a9ec35f which will compile error even on iOS side.
`Undefined symbols for architecture arm64: "cocos2d::cocos2dVersion()", referenced from: __GLOBAL__sub_I_Tracking.cpp in sdkbox(Tracking.o) ld: symbol(s) not found for architecture arm64`
Everything work properly 7dcffee  before the name changes, I have tried several ways but did not succeed. Any ideas?

**Comment by qoozta** [comment, marked as answer]
After changing the namespace back to cocos2d, everything work again.

**Reply by halx99** [reply]
You can add follow code at end of `axys.cpp` without engine namespace change.

```cpp
 namespace cocos2d {
     AX_DLL const char* cocos2dVersion()
     {
         return axysVersion();
     }
 }
```

**Reply by qoozta** [reply]
> You can add follow code at end of `axys.cpp` without engine namespace change.
> 
> ```c++
>  namespace cocos2d {
>      AX_DLL const char* cocos2dVersion()
>      {
>          return axysVersion();
>      }
>  }
> ```

This is not working, the pre-compiled library (SDKBox) not importing axys.cpp at all.

**Reply by halx99** [reply]
Another solution define namespace cocos2d as macro to ax

```
// in cocos2d.h
#define cocos2d ax

// in axys.cpp
#undef cocos2d
namespace cocos2d {
     AX_DLL const char* cocos2dVersion()
     {
         return axysVersion();
     }
 }
```

**Reply by qoozta** [reply]
> Another solution define namespace cocos2d as macro to ax
> 
> ```
> // in cocos2d.h
> #define cocos2d ax
> 
> // in axys.cpp
> #undef cocos2d
> namespace cocos2d {
>      AX_DLL const char* cocos2dVersion()
>      {
>          return axysVersion();
>      }
>  }
> ```

Just tested, same error. Never mind, SDKBox will soon be deprecated since Google Play required Billing 5.0 in November.
Thanks for your help!
